### PR TITLE
New sys-kernel/asahi-kernel package

### PR DESCRIPTION
**Pull request by mrhh69**
- based on kernel-build.eclass
- includes a patch to rust_is_available
- bindgen patch + newer bindgen version because upstream gentoo has only up to 0.63.0 (and that is masked, even)

### CODE REVIEW

**Comment by jannau** [comment]
bindgen 0.62.0 works for me. I think even bindgen 0.56.0 would be fine as long as llvm-16 is not installed.

File: metadata/md5-cache/sys-kernel/asahi-kernel-6.4.0_p1 [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+EAPI=8
+HOMEPAGE=https://asahilinux.org
+IDEPEND=|| ( sys-kernel/installkernel-gentoo sys-kernel/installkernel-systemd-boot ) initramfs? ( >=sys-kernel/dracut-049-r3 )
+INHERIT=kernel-build
+IUSE=experimental savedconfig +initramfs test +strip
+KEYWORDS=~arm64
+LICENSE=GPL-2
+RDEPEND=virtual/libelf
+RESTRICT=!test? ( test ) test? ( userpriv ) arm? ( test )
+SLOT=6.4.0_p1
```
Comment (jannau) [inline metadata/md5-cache/sys-kernel/asahi-kernel-6.4.0_p1:13]: stale metadata cache file

File: sys-kernel/asahi-kernel/files/bindgen.patch [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+diff --git a/scripts/min-tool-version.sh b/scripts/min-tool-version.sh
+index 20d483ec6f5f..5b80c5d3a9f8 100755
+--- a/scripts/min-tool-version.sh
++++ b/scripts/min-tool-version.sh
+@@ -30,7 +30,7 @@ rustc)
+ 	echo 1.62.0
+ 	;;
+ bindgen)
+-	echo 0.56.0
++	echo 0.65.1
```
Comment (jannau) [inline sys-kernel/asahi-kernel/files/bindgen.patch:76]: is there anything in the patch whice requires bindgen 0.65.1?

**Review by jannau — COMMENTED** [review]
I would expect that the kernel post install runs `update-m1n1` since the newly installed kernel might not boot in the case of incompatible device tree changes.

File: sys-kernel/asahi-kernel/files/bindgen.patch (hunk 2 above)
Comment (mrhh69) [reply sys-kernel/asahi-kernel/files/bindgen.patch:76]: the min version is 0.65.1 in the patch, so `make rustavailable` fails
older versions might be supported, I'll try 0.62.0

**Comment by mrhh69** [comment]
One thing I have yet to decide on is versioning, I see two options:

1. Use same versioning as in asahi-sources package: `maj.min.0-tag-asahi*`
2. Use versioning expected in kernel-{build,install}.eclass: `${PV}-asahi*` == `maj.min.0_ptag-asahi*`

The first approach is what I've already tried, but it would take alot more wrangling to actually implement correctly

**Comment by chadmed** [comment]
I think for versioning, copying what `asahi-sources` does is the way to go. The kernel ships with a `10-localversion.conf` that sets the version string to `maj.min.0-asahi-tag` anyway iirc

File: sys-kernel/asahi-kernel/files/rust_is_available.patch
Comment (AmateurECE) [inline sys-kernel/asahi-kernel/files/rust_is_available.patch:1]: Just a head's up, this was recently patched in upstream 🙂 https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=9eb7e20e0c5cd069457845f965b3e8a7d736ecb7

**Pull request closed without merging**